lib/veil/cipher/v2.rb

```diff
@@ -0,0 +1,41 @@
+require "base64"
+require "openssl"
+
+module Veil
+  class Cipher
+    class V2
+      attr_reader :key, :iv, :cipher
+
+      def initialize(opts = {})
+        @cipher = OpenSSL::Cipher.new("aes-256-cbc")
+        @key    = opts[:key] ? Base64.strict_decode64(opts[:key]) : cipher.random_key
+        @iv     = opts[:iv] ? Base64.strict_decode64(opts[:iv]) : cipher.random_iv
+      end
+
+      def encrypt(plaintext)
+        c = cipher
+        c.encrypt
+        c.key = key
```

Comment:
Can we do these assignments in `initialize` or are these split out so that we can test it more easily?